test/functional/wallet_resendwallettransactions.py

```diff
@@ -74,6 +79,41 @@ def run_test(self):
         node.setmocktime(now + 36 * 60 * 60 + 600)
         peer_second.wait_for_broadcast([txid])
 
+        self.log.info("Chain of unconfirmed not-in-mempool txs are rebroadcast")
+        # Ideally we would add a transaction that would spend the existing transaction and also be
+        # in mapWallet before it. However mapWallet is a std::unsorted_map which means that it is
+        # hard for us to externally construct such a transaction. So we will ignore that fact and
+        # rely on the fact that because there are only two items in mapWallet, 50% of the time the
+        # child will be placed before the parent in mapWallet. The incorrect behavior that this
+        # test tries to catch should thus fail intermittently at a rate of 50%.
```

Comment:
I am a bit confused on the intermittent test failure occurring here, maybe I am missing something.

Isn't sorting the order by wallet insertion index is what this PR fixes? So why would it still fail to reboadcast if `child.txid() < parent.txid()`? 

Also, does it make sense to have some unit testing to ensure this behavior of the `GetSorted()` function itself?

Reply:
> Isn't sorting the order by wallet insertion index is what this PR fixes? So why would it still fail to reboadcast if `child.txid() < parent.txid()`?

The comment is talking about the code before this PR and how the test tries to catch it, not that this test should fail intermittently after the PR.

> Also, does it make sense to have some unit testing to ensure this behavior of the `GetSorted()` function itself?

There are already tests for the underlying `wtxOrdered` itself (or tests that end up testing that behavior, although perhaps not specifically).

Reply:
So that means the code should not fail intermittently?? I will double check again that I am testing the correct binary, but I thoughts that's expected behavior from the comment..

Reply:
It should only fail intermittently on master.